ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -481,6 +511,16 @@ class WebSocketService(
     val StreamPredicate(resolved, unresolved, fn, dbQuery) =
       Q.predicate(request, resolveTemplateId, lookupType)
 
+    val requestOffset = Q.requestOffset(request)
+
+    val startingOffset =
+      (offPrefix, requestOffset) match {
+        case (Some(a), Some(b)) => Some(domain.StartingOffset.ordering.min(a, b))
+        case (a, None) if a.isDefined => a
+        case (None, b) if b.isDefined => b
+        case _ => None
+      }
```

Comment:
Some duplication with the `requestOffset` method implementation itself. I'd like to improve how the comparison is done and factor it out. Suggestions?

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9847/commits/ae9d8f7d2dda39f946f5bdac6dc9eb1f0b28b02b